### PR TITLE
Fix for is_secret_disabled

### PR DESCRIPTION
**Pull request by ShahafBenYakir**
## Status
Ready

## Description
Added Sade fix for the secret detection

### CODE REVIEW

File: Tests/scripts/hook_validations/secrets.py [hunk 1]
```diff
@@ -336,14 +340,15 @@ def remove_false_positives(line):
...
-    skip_secrets = False
+def is_secrets_disabled(line, skip_secrets):
     if bool(re.findall(r'(disable-secrets-detection)', line)):
         skip_secrets = True
     elif bool(re.findall(r'(disable-secrets-detection-start)', line)):
         skip_secrets = True
     elif bool(re.findall(r'(disable-secrets-detection-end)', line)):
         skip_secrets = False
+    elif not skip_secrets:
+        skip_secrets = False
```
Comment (yardensade) [inline Tests/scripts/hook_validations/secrets.py]: Delete 350+351

File: Tests/scripts/hook_validations/secrets.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
     import PyPDF2
 except ImportError:
     import pip._internal as pip
+
```
Comment (yardensade) [inline Tests/scripts/hook_validations/secrets.py]:
```suggestion
```

File: Tests/scripts/hook_validations/secrets.py [hunk 3]
```diff
@@ -29,27 +30,29 @@
...
+             r'[0-9]|25[0-5])\\.){3}(?:[0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))|(?:(?:[0-9A-Fa-f]' \
+             r'{1,4}:){,4}[0-9A-Fa-f]{1,4})?::(?:[0-9A-Fa-f]{1,4}:[0-9A-Fa-f]{1,4}|(?:(?:[0-9]|[1-9][0-9]' \
+             r'|1[0-9]{2}|2[0-4][0-9]|25[0-5])\\.){3}(?:[0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5]))|' \
+             r'(?:(?:[0-9A-Fa-f]{1,4}:){,5}[0-9A-Fa-f]{1,4})?::[0-9A-Fa-f]{1,4}|' \
              r'(?:(?:[0-9A-Fa-f]{1,4}:){,6}[0-9A-Fa-f]{1,4})?::)'
 IPV4_REGEX = r'\b(?:(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.){3}(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\b'
 DATES_REGEX = r'((\d{4}[/.-]\d{2}[/.-]\d{2})[T\s](\d{2}:?\d{2}:?\d{2}:?(\.\d{5,10})?([+-]\d{2}:?\d{2})?Z?)?)'
 # false positives
 UUID_REGEX = r'([\w]{8}-[\w]{4}-[\w]{4}-[\w]{4}-[\w]{8,12})'
+
```
Comment (yardensade) [inline Tests/scripts/hook_validations/secrets.py]:
```suggestion
```

File: Tests/scripts/hook_validations/secrets.py [hunk 4]
```diff
@@ -29,27 +30,29 @@
...
+
```
Comment (yardensade) [inline Tests/scripts/hook_validations/secrets.py]:
```suggestion
```

**Comment by ShahafBenYakir** [comment]
@yardensade thank you much work was done

**Pull request merged**